### ISSUE TITLE
Multiline class declaration with superclass + interface: Interface gets moved to newline

### ISSUE THREAD

**Issue by carstenhag** [pinterest/ktlint#1713]
## Expected Behavior
```
class HeaderContentViewModel @Inject constructor(
    mapService: IMapService,
    xyz: Xyz
) : FillingStationDetailsViewModel(
    mapService,
    xyz,
), IHeaderContentViewModel {
```
is accepted formatting.

https://kotlinlang.org/docs/coding-conventions.html#class-headers mentions similar cases but not this exact one. Keeping it in the same line seems to be the most reasonable solution to me. 

```
class Person(
    id: Int,
    name: String,
    surname: String
) : Human(id, name),
    KotlinMaker { /*...*/ }
```

## Observed Behavior
`Missing newline after "," (wrapping)` is reported. `ktLintFormat` formats it to 

```
class HeaderContentViewModel @Inject constructor(
    mapService: IMapService,
) : FillingStationDetailsViewModel(
    mapService,
),
    IHeaderContentViewModel {
```
which is not that pretty imo :D

## Steps to Reproduce
Format the above

## Your Environment

* Version of ktlint used: 0.47.1
* Relevant parts of the `.editorconfig` settings:
```
[*.{kt,kts}]
indent_size=4
insert_final_newline=true
max_line_length=120
ktlint_disabled_rules=trailing-comma-on-declaration-site,trailing-comma-on-call-site
```
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): 
* Version of Gradle used (if applicable): 7.5.1
* Operating System and version: macOS 12.3

**Comment by paul-dingemans** [comment]
I agree that code below does not look well formatted:
```
class HeaderContentViewModel @Inject constructor(
    mapService: IMapService,
) : FillingStationDetailsViewModel(
    mapService,
),
    IHeaderContentViewModel {
   // do something
}
```

The format you prefer was actually the format until and including `0.40.x` (I have not analysed why it was changed). 
```
class HeaderContentViewModel @Inject constructor(
    mapService: IMapService,
) : FillingStationDetailsViewModel(
    mapService,
), IHeaderContentViewModel {
   // do something
}
```
Althought this format looks a bit better, it is less consistent with how other argument lists are formatted.

Based on example below (see https://kotlinlang.org/docs/coding-conventions.html#class-headers):
```
class MyFavouriteVeryLongClassHolder :
    MyLongHolder<MyFavouriteVeryLongClass>(),
    SomeOtherInterface,
    AndAnotherOne
{
    fun foo() { /*...*/ }
}
```
I would say that the code should have been formatted as follows:
```
class HeaderContentViewModel(
    mapService: IMapService,
    xyz: Xyz,
) :
    FillingStationDetailsViewModel(
        mapService,
        xyz
    ),
    IHeaderContentViewModel
{
    // do something
}
```

IntelliJ IDEA default formatting however joins the opening brace `{` with last super type, resulting in code below:
```
class HeaderContentViewModel(
    mapService: IMapService,
    xyz: Xyz,
) :
    FillingStationDetailsViewModel(
        mapService,
        xyz
    ),
    IHeaderContentViewModel {
    // do something
}
```

This last code sample is also accepted by KtLint '0.47.x` and current snapshot for `0.48.x`. So for now, my suggestion would be to follow that format.

**Comment by paul-dingemans** [comment]
Related to #1349

**Comment by paul-dingemans** [comment]
The new `class-signature` (#1349, #875) rewrites the code sample below:
```
class HeaderContentViewModel @Inject constructor(
    mapService: IMapService,
    xyz: Xyz
) : FillingStationDetailsViewModel(
    mapService,
    xyz,
), IHeaderContentViewModel {
    // body
}
```

with code style `ktlint_official` to:
```
class HeaderContentViewModel
    @Inject
    constructor(
        mapService: IMapService,
        xyz: Xyz,
    ) : FillingStationDetailsViewModel(
            mapService,
            xyz,
        ),
        IHeaderContentViewModel {
    // body
}
```

with code styles `intellij_idea` and `android_studio` the format is kept unchanged and less optiomal as:
```
class HeaderContentViewModel @Inject constructor(
    mapService: IMapService,
    xyz: Xyz,
) : FillingStationDetailsViewModel(
    mapService,
    xyz,
),
    IHeaderContentViewModel {
    // body
}
```

**Comment by paul-dingemans** [comment]
See #2119
